go-controller/pkg/ovn/controller/services/repair.go

```diff
@@ -0,0 +1,113 @@
+package services
+
+import (
+	"time"
+
+	"github.com/pkg/errors"
```

Comment:
i think a reconciliation loop is something that we need to discuss and design for all resources in ovn-kubernetes, not just services. Can we please remove this from the PR and discuss it in another?

Reply:
this is needed for the upgrades, so there are not stale loadbalancer, it is better that each controller handles its own data to avoid dependencies

Reply:
I agree with you, but that's not the current design pattern we use in ovn-kubernetes today. Today the sync method is used to remove any stale OVN entities for a kubernetes resource.

Reply:
If this service controller run, all the all the other service and endpoints handlers are disabled to avoid any problems of concurrency or multiples things trying to modify resources.
Services and endpoints are only managed in the controller, maybe I´m not getting your point, sorry :thinking: